### PR TITLE
[tf] switch on separate storage device

### PR DESCRIPTION
**Pull request by rustielin**
## Motivation

Fullnodes were getting malformed logrotate configs due to missing device. This PR adds additional checks for `/dev/nvme1n1`. We can now be in one of four states: with or without `/dev/nvme1n1` and with or without persistent storage.

### Have you read the [Contributing Guidelines on pull requests](https://github.com/libra/libra/blob/master/CONTRIBUTING.md#pull-requests)?

yes

## Test Plan

tf apply dev config, SSH into fullnode. Fullnodes are non-persistent and do not have extra storage device, so it will resort to symlink with docker data-dir:

```
[root:validator@ip-10-0-4-147 /]$ ls -l
total 68
lrwxrwxrwx   1 root root     7 Apr 27 20:52 bin -> usr/bin
dr-xr-xr-x   4 root root  4096 Apr 27 20:53 boot
lrwxrwxrwx   1 root root    23 May 18 19:24 data -> /var/lib/docker/volumes
...

[root:validator@ip-10-0-4-147 /]$ ls /data
ecs-rustietest-fullnode-1-64-libra-data-9484e9f097fbde986b00  metadata.db  README
```

Logrotate looking good:

```
[root:validator@ip-10-0-4-147 /]$ cat /etc/logrotate.d/libra
hourly
/data/*/_data/libra.log {
        maxsize 500M
        rotate 100
        compress
        delaycompress
        copytruncate
}

/data/*/_data/libra_structlog.log {
        maxsize 500M
        rotate 100
        compress
        delaycompress
        copytruncate
}
```

SSH into validator. Validators are non-persistent but have an extra storage device, so we can mount that to docker data-dir to take advantage of the extra faster device:

```
[ec2-user:validator@ip-10-0-10-159 /]$ df
Filesystem     1K-blocks    Used Available Use% Mounted on
devtmpfs         1897664       0   1897664   0% /dev
tmpfs            1911140       0   1911140   0% /dev/shm
tmpfs            1911140     600   1910540   1% /run
tmpfs            1911140       0   1911140   0% /sys/fs/cgroup
/dev/nvme0n1p1  30830568 3151960  27347672  11% /
/dev/nvme1n1    30832548  921896  28321404   4% /data
tmpfs             382232       0    382232   0% /run/user/1000

...

[ec2-user:validator@ip-10-0-10-159 /]$ ls /data
ecs-rustietest-validator-2-61-libra-data-80bcdcddf6bd969c8101  lost+found  README
```

Logrotate looking good:

```
[ec2-user:validator@ip-10-0-10-159 /]$ cat /etc/logrotate.d/libra
hourly
/data/*/_data/libra.log {
        maxsize 500M
        rotate 100
        compress
        delaycompress
        copytruncate
}

/data/*/_data/libra_structlog.log {
        maxsize 500M
        rotate 100
        compress
        delaycompress
        copytruncate
}
```

## Related PRs

(If this PR adds or changes functionality, please take some time to update the docs at https://github.com/libra/website, and link to your PR here.)

### CODE REVIEW

File: terraform/templates/ec2_user_data.sh [hunk 1]
```diff
@@ -2,6 +2,9 @@
 # Copyright (c) The Libra Core Contributors
 # SPDX-License-Identifier: Apache-2.0
 
+export log_path="/data/libra/${host_log_path}"
```
Comment (sausagee) [inline terraform/templates/ec2_user_data.sh:5]: Will this override the configs in validator.tf and variable.tf?

Comment (rustielin) [reply terraform/templates/ec2_user_data.sh:5]: Nope it takes input from the template_file in `validators.tf` and renders it. https://github.com/libra/libra/blob/317966efb1007eb1afa9716e0b56c7508fc16727/terraform/validators.tf#L134

The rendered user_data template_file is shared by both `validators.tf` and `fullnodes.tf`

Comment (rustielin) [reply terraform/templates/ec2_user_data.sh:5]: logs can be in one of two places: /data/libra for persistent storage, or /data/<container name>/_data

**Pull request closed without merging**